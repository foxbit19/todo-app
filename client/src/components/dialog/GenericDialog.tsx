import React, { ReactElement } from 'react'
import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, Typography } from '@mui/material'

interface Props {
    open: boolean,
    title: string,
    text: string,
    children?: ReactElement,
    button?: ReactElement,
    onClose?: () => void
}

const GenericDialog = (props: Props) => {
    return <Dialog open={props.open} onClose={props.onClose} maxWidth={'lg'}>
        <DialogTitle>{props.title}</DialogTitle>
        <DialogContent>
            <DialogContentText>
                {props.text}
            </DialogContentText>
            {props.children}
        </DialogContent>
        <DialogActions>
            <Button data-testid='close_dialog_button' onClick={props.onClose}>Close</Button>
            {props.button}
        </DialogActions>
    </Dialog>
}

export default GenericDialog