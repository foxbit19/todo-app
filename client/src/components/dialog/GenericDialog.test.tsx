import React from 'react';
import { render, screen } from '@testing-library/react';
import sinon from 'sinon';
import userEvent from '@testing-library/user-event';
import Item from '../../models/item';
import GenericDialog from './GenericDialog';

describe('generic dialog', () => {
    let item: Item

    beforeEach(() => {
        item = new Item(1, 'test', 1)
    })

    test('it matches snapshot', () => {
        const view = render(<GenericDialog open={true} title='' text='' />);
        expect(view).toMatchSnapshot();
    });

    test('it calls onClose when close button is clicked', () => {
        const handleClick = sinon.spy()
        render(<GenericDialog open={true} title='' text='' onClose={handleClick} />);

        userEvent.click(screen.getByTestId('close_dialog_button'))

        expect(handleClick.called).toBeTruthy();
    });

    test('it show title and text', () => {
        render(<GenericDialog open={true} title='My title' text='My text' />);

        expect(screen.getByText('My title')).toBeTruthy();
        expect(screen.getByText('My text')).toBeTruthy();
    });
})