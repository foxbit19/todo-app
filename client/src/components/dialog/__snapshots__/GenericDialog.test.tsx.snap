// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic dialog it matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r0:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthLg css-1fu2e3p-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
            id=":r0:"
          />
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <p
              class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
            />
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              data-testid="close_dialog_button"
              tabindex="0"
              type="button"
            >
              Close
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
